typing/typecore.ml

```diff
@@ -5876,6 +5974,32 @@ and type_function_cases_expect
     cases, partial, ty_fun
   end
 
+and type_effect_cases
+    : type k . k pattern_category -> _ -> _ -> _ -> Parsetree.case list -> _
+               -> k case list
+  = fun category env ty_res_explained loc caselist conts ->
+      let { ty = ty_res; explanation = _ } = ty_res_explained in
+      let _ = newvar () in
+      (* remember original level *)
+      with_local_level begin fun () ->
+        (* Create a locally type abstract type for effect type. *)
+        let new_env, ty_arg, ty_cont =
+          let decl = Ctype.new_local_type ~loc Definition in
+          let scope = create_scope () in
+          let name = Ctype.get_new_abstract_name env "%eff" in
+          let id, new_env = Env.enter_type ~scope name decl env in
```

Comment:
```diff
-    let id, new_env = Env.enter_type ~scope name decl env in
+    let id = Ident.create_scoped ~scope name in
+    let new_env = Env.add_type ~check:false id decl env in
```
to avoid the unused type declaration warning, which makes sense for an implicitly added type declaration.
Sorry for introducing the bug.

Reply:
Let me make the change and push a fix.